### ISSUE TITLE
Testing on Different Images

### ISSUE THREAD

**Issue by nbibin001** [MaybeShewill-CV/lanenet-lane-detection#508]
Hi, first of all thanks a lot for sharing this repository and detailed instructions for setting it up. It was very helpful to replicate and test it with the images in the data\tusimple_test_image folder. And with the default images present it worked just fine and I could replicate the results. But then I was trying to test it on some of the images from a different data set, it is not really working as expected. I am not seeing the detected lanes overlayed on the source image. I was just curious  Is there something additional I need to do, or is it the limitation of the model, and I have to further train the model (transfer learning) to generalize it for my data set as well ? Really appreciate your help and thoughts ... 

![image](https://user-images.githubusercontent.com/15659590/120089277-a0470a80-c0c6-11eb-9a55-6df10cf9dde2.png)

**Comment by MaybeShewill-CV** [comment]
@nbibin001 The postprocess script was only suitable for tusimple dataset you need to adjust it according to you own data:)

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV can you please tell me how can I proceed with getting lane coordinates and then evaluating different performance metrics if I have only binary segmented output not instance segmented one. Thanks.

**Comment by MaybeShewill-CV** [comment]
@akshatbhutra You can only get all lane pixels via binary segmentation result. To obtain different lanes you need to do cluster yourself:)

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV Thanks for fast reply :). I am interested in saving lane coordinates of each lane instance separately in a JSON file so that finally I can use this official tusimple evaluation script: https://github.com/TuSimple/tusimple-benchmark/blob/master/evaluate/lane.py but the problem is I am doing semantic segmentation, not the instance. Any reference code for that?

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV can you please give some reference for clustering? Thanks.

**Comment by MaybeShewill-CV** [comment]
@akshatbhutra You may google some traditional lane detection method to get some help:)

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV is it now possible to print the summary of layers of LaneNet Model?

**Comment by MaybeShewill-CV** [comment]
@MaybeShewill-CV Tensorflow 2 was not supported here:)

**Comment by akshatbhutra** [comment]
Okay , Thanks for the reply. Can you please tell me how I could know (from which script) about the the two branch architecture (encoder-decoder part) mentioned in paper in more details (like how layers are stacked one after another?). Thanks!

**Comment by MaybeShewill-CV** [comment]
@nbibin001 Maybe try tensorboard :)

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV Is there any reason why [official tusimple evaluation script](https://github.com/TuSimple/tusimple-benchmark/blob/master/evaluate/lane.py) was not used? Thanks!

**Comment by MaybeShewill-CV** [comment]
@akshatbhutra I do not know the reason. Maybe you can get some help on tusimple's website:)

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV Do you know any reference for the implementation of this architecture using Keras? Thanks!

**Comment by MaybeShewill-CV** [comment]
@akshatbhutra One branch is for semantic segmentation the other if for instance segmentation:)

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV How many stages encoder share between the two branches: first two (as mentioned in the paper) or first four (as mentioned in [this script](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/master/semantic_segmentation_zoo/vgg16_based_fcn.py). Thanks!

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV can you please explain the purpose of lines 299-305 of [this script](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/master/semantic_segmentation_zoo/vgg16_based_fcn.py). Thanks!

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV
Any thoughts on the following questions regarding [this script](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/master/semantic_segmentation_zoo/vgg16_based_fcn.py).
(1) What lines 299-305 are doing?

(2) Why in line 309 out_channel = self._class_nums = self._cfg.DATASET.NUM_CLASSES = 2? Shouldn't it be 1 as the binary segmentation branch output a one
channel image? 

(3) In line 342  why out_channels_nums=64 for the last layer of decoder block  of instance segmentation? Should not it be 4 (N) or am I missing something? (as the last layer of the embedding branch outputs an N-channel image, with N the embedding dimension.)
Thanks a lot!

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV Could you please tell me am  I referring to [the correct script](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/master/semantic_segmentation_zoo/vgg16_based_fcn.py) to debug the encoder-decoder architecture layer by layer?

**Comment by akshatbhutra** [comment]
@MaybeShewill-CV can you please shed some light on the meaning of the terms "binary_seg_logits" and "binary_seg_label". Thanks!

**Comment by MaybeShewill-CV** [comment]
@akshatbhutra You may look up cross entropy loss's api for help

**Comment by akshatbhutra** [comment]
Okay, thanks!
